ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiWorldStateProvider.java

```diff
@@ -338,6 +338,9 @@ public TrieLogManager getTrieLogManager() {
   @Override
   public void setArchiveStateUnSafe(final BlockHeader blockHeader) {
```

Comment:
this method name should change to reflect its use.  IMO it isn't setting the archive unsafe for use like the name implies, but rather is resetting the archive cache and adding the new pivot as the only entry.  Perhaps `resetArchiveStateTo(BlockHeader)`

Similarly, BonsaiWorldState.setArchiveUnsafe should be something more along the lines of `resetWorldStateTo(BlockHeader)`

These methods could use javadoc describing the intent

Reply:
updated the code